pubnub-gson/src/test/kotlin/com/pubnub/internal/v2/PNConfigurationImplTest.kt

```diff
@@ -0,0 +1,41 @@
+package com.pubnub.internal.v2
+
+import com.pubnub.api.UserId
+import com.pubnub.api.v2.PNConfiguration
+import com.pubnub.internal.BasePubNubImpl
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class PNConfigurationImplTest {
+    @Test
+    fun testDefaultTimeoutValues() {
+        val config = PNConfiguration.builder(UserId(BasePubNubImpl.generateUUID()), "demo")
+        assertEquals(300, config.presenceTimeout)
+        assertEquals(0, config.heartbeatInterval)
+    }
+
+    @Test
+    fun testCustomTimeoutValues1() {
+        val config = PNConfiguration.builder(UserId(BasePubNubImpl.generateUUID()), "demo")
+        config.setPresenceTimeout(100)
```

Comment:
Is this possible to do:
   val config = PNConfiguration.builder(UserId(BasePubNubImpl.generateUUID()), "demo")  
                                    .presenceTimeout(100)
                                    .heartbeatInterval(21)
                                    .retryConfiguration(new RetryConfiguraiton.Linear())
?

Reply:
ah, so that is a good question - I followed the convention from the old PNConfiguration and prefixed the methods with `set` and allowed chaining, so you can do:
```
PNConfiguration.builder(UserId(BasePubNubImpl.generateUUID()), "demo")
.setPresenceTimeout(100)
.setHeartbeatInterval(21)
.setRetryConfiguration(new RetryConfiguraiton.Linear())
```

some of our other builders use setters without `set`
the question is which should we use?
I'm leaning towards setX for PNConfiguration.Builder because the old PNConfiguration had it, so it will be easy to copy/paste old code and simply change the class to Builder... but I'm open to thoughts

Reply:
I think in Java SDK we have convention without set e.g.
```
        pubNub.publish()
                .message(messagePayload)
                .channel(expectedChannel)
                .sync()
```

```
        pubnub.subscribe()
                .channels(Arrays.asList(channels))
                .withPresence()
                .execute();
```

I see that for old customers having setters without `set` might require more effort but for new customers API will be consistent I think.

What do you think?

Reply:
yeah it might make sense to make it consistent, I'll remove the `set` prefix

Reply:
done